### PR TITLE
fix(cloudfront): cannot use same EdgeFunction in multiple stacks

### PR DESCRIPTION
**Pull request by jogold**
Using the same `EdgeFunction` in multiple stacks correctly created
multiple stacks in `us-east-1` but with the same SSM parameter name. As
a consquence, only one stack could be deployed.

Fix it by including the stack unique address in the SSM parameter name.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
Tested to work for existing stacks.

```bash
git checkout master && yarn build && yarn integ integ.distribution-lambda-cross-region.js --no-clean && git checkout edge-fn-ssm-param-name && yarn build && yarn integ integ.distribution-lambda-cross-region.js --no-clean
```

**Review by njlynch — COMMENTED** [review]
Looks great, except for one minor ID readability nitpick. What do you think?

**Comment by jogold** [comment]
> Looks great, except for one minor ID readability nitpick. What do you think?

How about this?

```ts
const parameterName = `${parameterNamePrefix}${Stack.of(this).stackName}${id}`;
```

**Comment by njlynch** [comment]
> How about this?
> 
> ```ts
> const parameterName = `${parameterNamePrefix}${Stack.of(this).stackName}${id}`;
> ```

I can live with that, I suppose.

Totally agreed the stack ID could be friendlier too.

**Comment by jogold** [comment]
> Totally agreed the stack ID could be friendlier too.

Do you want to change it in this PR? It will leave "orphaned" stacks for customers already using the construct. On the other hand this construct is still marked as experimental... up to you!

**Comment by jogold** [comment]
Maybe the best option is to do:

```ts
const parameterNamePrefix = '/EdgeFunctionArn';
const parameterName = `${parameterNamePrefix}/${this.node.path}`;
```
(using slashes is common for SSM parameters)

otherwise deploying different `EdgeFunction`s with the same id but defined in different constructs will also fail.

**Comment by njlynch** [comment]
What about multiple stacks in different regions? If I take the exact same (environment-agnostic) stack, and deploy it to ap-south-1 and eu-west-1, I would get the same stack name + node addr from both, and still hit a parameter name conflict. We could incorporate the current region in the name as well to address that.

File: packages/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts [hunk 1]
```diff
@@ -148,8 +148,8 @@ export class EdgeFunction extends Resource implements lambda.IVersion {
 
   /** Create a support stack and function in us-east-1, and a SSM reader in-region */
   private createCrossRegionFunction(id: string, props: EdgeFunctionProps): FunctionConfig {
-    const parameterNamePrefix = 'EdgeFunctionArn';
-    const parameterName = `${parameterNamePrefix}${id}`;
+    const parameterNamePrefix = '/EdgeFunctionArn';
```
Comment (jogold) [inline packages/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts:151]: maybe `/cdk/EdgeFunctionArn` here?

Comment (njlynch) [reply packages/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts:151]: Sure, I can see that. I'd be happy with this as-is, but open to the `/cdk` prefix. Do you want to update?

**Review by njlynch — APPROVED** [review]

**Pull request merged**